pylink/jlink.py

```diff
@@ -2192,9 +2192,18 @@ def flash(self, data, addr, on_progress=None, power_on=False, flags=0):
         except errors.JLinkException:
             pass
 
-        res = self.flash_write(addr, data, flags=flags)
+        # Perform read-modify-write operation.
+        res = self._dll.JLINKARM_BeginDownload(flags=flags)
+        if res < 0:
+            raise errors.JLinkEraseException(res)
 
-        return res
+        bytes_flashed = self._dll.JLINKARM_WriteMem(addr, len(data), data)
```

Comment:
Debugging, it looks like `data` has to be a `bytes` vs. a list of integers, so if `data` is a list of integers, this will fail. I think that means we have to call `bytes(data)` here.
- [ ] Convert `data` to bytes if not.

Reply:
Do you want to retain the signature (list of bytes) and convert to `bytes` when needed?

Reply:
Seems to be working fine here.